pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -296,15 +298,301 @@ func (m *kubeGenericRuntimeManager) GetPods(all bool) ([]*kubecontainer.Pod, err
 	return result, nil
 }
 
-// SyncPod syncs the running pod into the desired pod.
-func (m *kubeGenericRuntimeManager) SyncPod(pod *api.Pod, _ api.PodStatus,
-	podStatus *kubecontainer.PodStatus, pullSecrets []api.Secret,
-	backOff *flowcontrol.Backoff) (result kubecontainer.PodSyncResult) {
-	result.Fail(fmt.Errorf("not implemented"))
+// podContainerSpecChanges keeps information on changes that need to happen for a pod.
+type podContainerSpecChanges struct {
+	// Whether need to create a new sandbox.
+	CreateSandbox bool
+	// The id of existing sandbox if not needed to create a new one.
+	SandboxID string
+	// The attempt number of creating sandboxes for the pod.
+	Attempt uint32
+
+	ContainersToStart map[int]string
```

Comment:
Need to document what these two fields are, what the int and the string represents.